### ISSUE TITLE
[icetray] duplicate code in I3Tray::Execute overloads (Trac #1420)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1412]
The two overloads look very similar, with lots of identical code. Let's clean that up so we only have one copy.

Migrated from https://code.icecube.wisc.edu/ticket/1420
```json
{
    "status": "closed", 
    "changetime": "2019-01-11T20:39:50", 
    "description": "The two overloads look very similar, with lots of identical code. Let's clean that up so we only have one copy.", 
    "reporter": "david.schultz", 
    "cc": "olivas, nega", 
    "resolution": "fixed", 
    "_ts": "1547239190325916", 
    "component": "combo core", 
    "summary": "[icetray] duplicate code in I3Tray::Execute overloads", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2015-11-04T20:01:36", 
    "milestone": "", 
    "owner": "olivas", 
    "type": "enhancement"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20151104T20:03:41: 
* olivas changed owner from "" to "olivas"
* olivas changed status from "new" to "accepted"

**Comment by icecube-trac** [comment]
Trac update at 20151104T20:32:24: 
* olivas changed resolution from "" to "fixed"
* olivas commented:

> In changeset 2338:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2338"
> removed duplicate code in I3Tray::Execute.  nearly all of it was duplicated with the exception of two lines.  one of which we'd actually want in both...which is why we don't duplicate code. (DRY => Don't Repeat Yourself).  closes #1412.
> ```


* olivas changed status from "accepted" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20160318T21:14:10: 
* olivas commented:

> Milestone offline V15-12-00 deleted


* olivas changed milestone from "offline V15-12-00" to ""

**Comment by icecube-trac** [comment]
Trac update at 20190111T20:39:50: nwhitehorn commented:

> In changeset 168671:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="168671"
> removed duplicate code in I3Tray::Execute.  nearly all of it was duplicated with the exception of two lines.  one of which we'd actually want in both...which is why we don't duplicate code. (DRY => Don't Repeat Yourself).  closes #1412.
> ```
